### DISCUSSION TITLE
Example of users in groups?

### DISCUSSION THREAD

**Discussion by richardltc** [pocketbase/pocketbase#165] [Q&A] [answered]
Could we have an example of authenticating a user that's in a team? Effectively, I want to make sure that only users in a particular team/group are able to add/edit posts? Is it only possible to have them as "admins" at the moment?

**Comment by ganigeorgiev** [comment, marked as answer]
There are several ways you can do that but in general you'll have to create a new user profile field.

#### v1. If your team/group/role has only a string/number identifier (eg. `manager`, `regular_user`, etc.)

- Go to "Users > Edit profile collection" and **create a new text or multiple select field**.
- Navigate to the API rules to your collections that you want to restrict and define the following rule:
  `@request.user.id != "" && @request.user.profile.group = "manager"`
  The above will allow only authenticated users with `profile.group = "manager"` (where "manager" is one of the possible group values)
  A similar example is shown here - https://pocketbase.io/docs/manage-users/#custom-fields.

#### v2. If your team/group/role has some additional info like name, description, etc.
- Create a new collection `group` with`name`, `description` and whatever other fields you need.
-  Go to "Users > Edit profile collection" and **create a  new relation field** named **"group"** pointing to the `group` collection from the previous step.
- In your `posts` collection you could use the following rule:
  `@request.user.id != "" && @request.user.profile.group = @collection.group.id && @collection.group.name = "manager"`

Since the user profile is also a regular collection, you may want to set a custom API rules in the profiles collection to prevent users from changing their team/group on their own (by default only admins and the user owner can edit the user profile).

> The rule in the second example will be simplified in future versions and you will be able to query through the user profile relation fields directly without requiring you to add the `group` join (aka. `@collection.group.id`), it will look like this:
`@request.user.id != "" && @request.user.profile.group.name = "manager"`

**Reply by ganigeorgiev** [reply]
@richardltc Just to let you know that I've added in master support for filtering `@request.user.profile.*` relation fields, or in other words, the following:
`@request.user.id != "" && @request.user.profile.group = @collection.group.id && @collection.group.name = "manager"`
can now be written as:
`@request.user.id != "" && @request.user.profile.group.name = "manager"`.

It will be available with the next minor release I'm planning for later this week.
